ca/certificates.go

```diff
@@ -209,6 +215,11 @@ func (rca *RootCA) RequestAndSaveNewCertificates(ctx context.Context, paths Cert
 		return nil, err
 	}
 
+	// FIXME: Is there a way to write *both* the new key and cert
+	// atomically? Maybe we change the credential storage model to have a
+	// single 'swarm-node.pem' bundle which contains the key PEM followed
+	// by the certificate PEM(s)?
```

Comment:
Let's remove this comment. It's a very valid concern, but we've discussed it before and have a strategy that works without combining the cert and the key.

The way it works is we first write the new key to a temporary file. Next, when we receive the signed certificate, we atomically overwrite the old certificate with it. Finally, we rename the temporary key file to replace the old key.

If there is some kind of crash before the cert is downloaded, it's as if we didn't do any certificate rotation, because only the new key exists, and it's in a temporary file that didn't overwrite anything.

If the crash happens after the certificate is downloaded but before the key is moved into place, this is also handled. When loading certificates from disk, we try using the temporary key file if the key at the normal path didn't work with the certificate.

This should be better documented. Feel free to add a comment explaining it.